src/ID.java

```diff
@@ -0,0 +1,12 @@
+public abstract class ID {
+    @NullableWarning
+    private Integer id;
```

Comment:
It is better to use primitives whenever its possible

Reply:
In this case, when I used int, id was zero, and when I used Integer id is null. So, I decided to use Integer.